### PR TITLE
Test and refactor whydoomed

### PR DESCRIPTION
**Pull request by iamtextbased**
I was working on the "[Include temperature indicator in scenario brief](https://github.com/MegaMek/mekhq/issues/2129)" issue on MekHQ and noticed that most/all of the temperature related code lived on Megamek's "PlanetaryConditions" class.

So I added tests to the section of the code responsible for extreme temperature detection and slightly refactored it so I can later call a "isExtremeTemperature" method on MekHQ to obtain that information.

This PR doesn't change any behaviour on Megamek.

### CODE REVIEW

**Review by Windchild292 — COMMENTED** [review]
Two style tidbits, while the tests look good from a once-over.

File: megamek/src/megamek/common/PlanetaryConditions.java [hunk 1]
```diff
@@ -618,10 +618,7 @@ public String whyDoomed(Entity en, IGame game) {
         if((windStrength == WI_STORM) && ((en instanceof Infantry) && !(en instanceof BattleArmor))) {
             return "storm";
         }
-        if ((temperature > 50 || temperature < -30) && en.doomedInExtremeTemp()) {
-            if (Compute.isInBuilding(game, en)) {
-                return null;
-            }
+        if (this.isExtremeTemperature() && en.doomedInExtremeTemp() && !Compute.isInBuilding(game, en)) {
```
[the remark this answers is not in the document]
Comment (iamtextbased) [reply megamek/src/megamek/common/PlanetaryConditions.java]: Done!

File: megamek/src/megamek/common/PlanetaryConditions.java [hunk 2]
```diff
@@ -851,6 +848,10 @@ public boolean isVacuum() {
         return (atmosphere == ATMO_VACUUM) || (atmosphere == ATMO_TRACE);
     }
 
+    public boolean isExtremeTemperature() {
+        return temperature > 50 || temperature < -30;
```
[the remark this answers is not in the document]
Comment (iamtextbased) [reply megamek/src/megamek/common/PlanetaryConditions.java]: Done!

File: megamek/unittests/megamek/common/PlanetaryConditionsTest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        IHex mockHex = Mockito.mock(Hex.class);
+        Coords mockCoords = Mockito.mock(Coords.class);
+
+        // Trace atmosphere - Entity doomed in vacuum/trace atmosphere
+
+        planetaryConditions = new PlanetaryConditions();
+        planetaryConditions.setAtmosphere(PlanetaryConditions.ATMO_TRACE);
+        Mockito.when(mockGame.getPlanetaryConditions()).thenReturn(planetaryConditions);
+        mockEntity = Mockito.mock(Infantry.class);
+        Mockito.when(mockEntity.doomedInVacuum()).thenReturn(true);
```
Comment (sixlettervariables) [inline megamek/unittests/megamek/common/PlanetaryConditionsTest.java:25]: I have been writing quite a few unit tests in MekHQ and have found adding a static import of `Mockito.*` really aids in the readability of these tests _and_ saves you hundreds (if not thousands) of keystrokes. I realize existing unit tests in MM don't quite follow that approach, but I figured I would pass it along.

**Review by sixlettervariables — APPROVED** [review]

**Comment by iamtextbased** [comment]
I noticed that to add the temperature to MekHQ's scenario brief I'd have to create a method "getTemperatureDisplayableName". Should I include it on this PR or should I create that on a new one to avoid mixing refactor/feature PRs?

**Comment by sixlettervariables** [comment]
This won't land until after stable, so feel free to add something like that.

File: megamek/unittests/megamek/common/PlanetaryConditionsTest.java (hunk 3 above)
Comment (iamtextbased) [reply megamek/unittests/megamek/common/PlanetaryConditionsTest.java]: Done!

**Review by sixlettervariables — APPROVED** [review]
Approved for 0.49

**Pull request merged**